packages/colorpickers/src/elements/ColorDialog/index.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, {
+  useState,
+  useRef,
+  Children,
+  cloneElement,
+  forwardRef,
+  ReactElement,
+  HTMLAttributes
+} from 'react';
+import PropTypes from 'prop-types';
+import mergeRefs from 'react-merge-refs';
+import { Button } from '@zendeskgarden/react-buttons';
+import { GARDEN_PLACEMENT } from '@zendeskgarden/react-modals';
+import Chevron from '@zendeskgarden/svg-icons/src/16/chevron-down-stroke.svg';
+import { ColorPicker, IColorPickerProps } from '../ColorPicker';
+import { IColor } from '../../utils/types';
+import {
+  StyledIcon,
+  StyledButton,
+  StyledFauxInput,
+  StyledDialogPreview,
+  StyledTooltipModal,
+  StyledTooltipBody
+} from '../../styled';
+
+export interface IColorDialogProps extends IColorPickerProps {
+  /**
+   * Handles close actions. Can be triggered from the backdrop.
+   *
+   * @param {Object} color a color picker state
+   */
+  onClose?: (color: IColor) => void;
+  /** Adjusts the placement of the color dialog */
+  placement?: GARDEN_PLACEMENT;
+  /** Disables the color dialog button */
+  disabled?: boolean;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLButtonElement>
+ */
+export const ColorDialog = forwardRef<
+  HTMLButtonElement,
+  IColorDialogProps & Omit<HTMLAttributes<HTMLButtonElement>, 'color' | 'onChange'>
+>(
+  (
+    { color, disabled, defaultColor, placement, onChange, onClose, labels, children, ...props },
+    ref
+  ) => {
+    const isControlled = color !== null && color !== undefined;
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    const colorPickerRef = useRef<HTMLDivElement>(null);
+    const mergedRef = mergeRefs([ref, buttonRef]);
+    const [referenceElement, setReferenceElement] = useState<HTMLButtonElement | null>();
+    const [uncontrolledColor, setUncontrolledColor] = useState<string | IColor>(
+      defaultColor || '#ffffff'
+    );
+
+    const onClick = () => {
+      if (referenceElement) {
+        setReferenceElement(null);
+      } else {
+        setReferenceElement(buttonRef.current);
+      }
+    };
+
+    return (
+      <>
+        {children ? (
+          cloneElement(Children.only(children as ReactElement), {
+            onClick,
+            ref: mergedRef
+          })
+        ) : (
+          <StyledFauxInput tabIndex={-1} disabled={disabled}>
+            <StyledButton disabled={disabled} ref={mergedRef} onClick={onClick} {...props}>
```

Comment:
This construct is problematic. This component needs to function as a Garden `<button>`. Right now, it behaves as some kind of hybrid between a `button` and a `select`. Functionally, it should behave just like https://garden.zendesk.com/components/menu#default – handling mouse and keyboard focus accordingly.

The `StyledButton` should receive all of the intended CSS to get the backround & borders right per designs. I would suggest that `StyledFauxInput` is not needed.